Fix openshift_kube_config_path defaults

mtnbikenc opened the pull request:
- Expand the path to full path using expanduser and realpath filters
- Move node upgrade tasks inside role to use the same defaults

Bug 1708568
Bug 1708588

vrutkovs commented on roles/openshift_node/defaults/main.yml:
```diff
@@ -1,5 +1,5 @@
 ---
-openshift_kubeconfig_path: '~/.kube/config'
+openshift_kubeconfig_path: "{{ openshift_kubeconfig_path | default('~/.kube/config') | realpath }}"
```
Why would we need `default(` here? These are role defaults, so user should be able to redefine those, aren't they?

mtnbikenc replied:
If the user doesn't provide anything, default to ~/.kube/config and expand to full path.
If the user does provide something, still pass it through realpath to expand to a full path.

mtnbikenc replied:
The problem is that `oc` doesn't deal with `~` so the role default doesn't work.

sdodson replied:
Actually, I'm worried that this may cause a recursive loop. Looking to see if we have any other examples of this pattern.

vrutkovs replied:
Why not using `realpath` in the task where `openshift_kubeconfig_path` is being used?

mtnbikenc replied:
Yep, busted. ...WIP

sdodson commented:
/lgtm

sdodson commented:
/lgtm

sdodson commented:
/lgtm cancel 
is what i mean :-(

mtnbikenc commented:
Updated to use `expanduser` filter which is actually the filter I wanted (expands '~').
Updated to create `openshift_node_kubeconfig_path` from `openshift_kubeconfig_path`

mtnbikenc commented:
Tested scaleup and upgrade with undefined `openshift_kubeconfig_path` (to use role default).
Both succeeded.
Removing WIP.

sdodson commented:
/lgtm

mtnbikenc commented:
/retest
```
Failing tests:

[Feature:DeploymentConfig] deploymentconfigs when run iteratively [Conformance] should only deploy the last deployment [Suite:openshift/conformance/parallel/minimal]
```

sdodson commented:
```
[sig-cli] Kubectl client [k8s.io] Simple pod should contain last line of the log [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] subPath should support existing directory [Suite:openshift/conformance/parallel] [Suite:k8s]
```
/override ci/prow/e2e-aws-scaleup-rhel7
